src/sagas.js

```diff
@@ -245,6 +282,80 @@ function* startVm (action) {
   yield stopProgress({ vmId: action.payload.vmId, name: 'start', result })
 }
 
+function* showEditVm (action) {
+  yield put(updateDialogType('edit'))
+  const cluster = Selectors.getClusterById(action.payload.vm.get('cluster').get('id'))
+  yield put(updateCluster(cluster))
+
+  const template = Selectors.getTemplateById(action.payload.vm.get('template').get('id'))
+  yield put(updateTemplate(template))
+
+  yield put(updateVmId(action.payload.vm.get('id')))
+
+  const os = Selectors.getOperatingSystemByName(action.payload.vm.get('os').get('type'))
+  yield put(updateOperatingSystem(os))
+
+  const name = action.payload.vm.get('name')
+  yield put(updateVmName(name))
+
+  const memory = action.payload.vm.get('memory').get('total')
+  yield put(updateVmMemory(memory))
+
+  const cpu = action.payload.vm.get('cpu').get('vCPUs')
+  yield put(updateVmCpu(cpu))
+  yield put(openVmDialog())
+  yield put(setVmDetailToShow({ vmId: action.payload.vm.get('id') }))
+}
+
+function* showAddNewVm (action) {
+  yield put(setVmDetailToShow({ vmId: '0' }))
+  yield put(updateDialogType('create'))
+  const cluster = Selectors.getFirstCluster()
+  yield put(changeCluster(cluster))
+  yield put(updateVmId('0'))
+  yield put(updateVmName(''))
+  yield put(openVmDialog())
+}
+
+function* handleClusterChange (action) {
+  yield put(updateCluster(action.payload.cluster))
+  // After every cluster change, set template to Blank
+  const blankTemplate = Selectors.getTemplateById('00000000-0000-0000-0000-000000000000')
+  yield put(changeTemplate(blankTemplate))
+}
+
+function* handleTemplateChange (action) {
```

Comment:
OK, it may turn out to be a pretty complicated change - can be done later. This patch brings lots of value also without this proposed change.